scripts/globals/dark_ixion.lua

```diff
@@ -462,3 +577,301 @@ xi.darkixion.onMobDisengage = function(mob)
+                    local xT = mobTarget:getXPos()
+                    local zT = mobTarget:getZPos()
+                    local xD = xT - xM
+                    local zD = zT - zM
+                    local away = { x = xM - xD, y = yM, z = zM - zD }
+
+                    mobArg:lookAt(away)
+                    mobArg:useMobAbility(2336) -- Acheron Kick
+                    mobArg:timer(1500, function(mobArg2)
+                        mobArg2:setBehaviour(0) -- in case the move is out-ranged
+                    end)
+                end
+
+                acheronKick(mob)
+                if mob:getLocalVar('double') == 1 then
+                    -- since no_turn is enabled, no need to change position again, just do it back to back
+                    mob:timer(3500, function(mobArg)
+                        acheronKick(mobArg)
+                    end)
+                end
+            elseif ws >= 3 then
+                -- Below handles the charge > zap logic, including if DI is glowing
+                -- zap -> lightning spear
+                -- zap2 -> Wrath of Zeus
+                local hitCount = mob:getLocalVar('double') == 1 and 2 or 1
+                if ws == 3 then
+                    mob:setLocalVar('zap', hitCount)
+                    mob:setLocalVar('zapTime', os.time() + 5)
+                else
+                    mob:setLocalVar('zap2', hitCount)
+                    mob:setLocalVar('zap2Time', os.time() + 5)
+                end
+
+                mob:setLocalVar('timeSinceWS', os.time() + 4)
+                mob:useMobAbility(2343) -- activate the glow first, then real WS
+            end
+        end
+    else
+        -- This is called after charging to either perform WoZ (zap2) or LS (zap)
+        if
+            mob:getLocalVar('zap2') >= 1 and
+            mob:getLocalVar('timeSinceWS') < os.time() - 1 and
+            os.time() >= mob:getLocalVar('zap2Time') and
+            mob:getAnimationSub() ~= 1
+        then
+            mob:setLocalVar('zap2', mob:getLocalVar('zap2') - 1)
+            mob:setTP(0)
+            mob:setBehaviour(xi.behavior.NO_TURN + xi.behavior.STANDBACK)
+            mob:useMobAbility(2334)
+            mob:setLocalVar('zap2Time', os.time() + 7)
+        end
+
+        if
+            mob:getLocalVar('zap') >= 1 and
+            mob:getLocalVar('timeSinceWS') < os.time() - 1 and
+            os.time() >= mob:getLocalVar('zapTime') and
+            mob:getAnimationSub() ~= 1
+        then
+            mob:setLocalVar('zap', mob:getLocalVar('zap') - 1)
+            mob:setTP(0)
+            mob:setBehaviour(xi.behavior.NO_TURN + xi.behavior.STANDBACK)
+
+            -- re-random lightning spear
+            local xM = mob:getXPos()
+            local zM = mob:getZPos()
+            local yM = mob:getYPos()
+            local xD = math.random(-4, 4)
+            local zD = math.random(-4, 4)
+            local away = { x = xM - xD, y = yM, z = zM - zD }
```

Comment:
You have this block as a similar looking pattern in a few places, break it out to a support function at the top of the file

Reply:
they are distinct, but I should be able to make a helper function that takes a parameter I suppose.